udfs/README.md

```diff
@@ -19,45 +19,120 @@ achieve feature parity in a migration from another data warehouse to BigQuery.
-shared datasets. Queries can then reference the shared UDFs via
+All UDFs within this repository are available under the `bqutil` project on
+publicly shared datasets. Queries can then reference the shared UDFs via
 `bqutil.<dataset>.<function>()`.
 
 ![Alt text](/images/public_udf_architecture.png?raw=true "Public UDFs")
 
-If you want to create and use the UDFs from this repository in your BigQuery dataset,
-you must provide your dataset name in the following `bqutil` command:
-
-`bq query --project_id YOUR_PROJECT_ID --dataset_id YOUR_DATASET_ID
---nouse_legacy_sql < SQL_FILE_NAME`
-
-Once you create the UDF in your BigQuery project, you can invoke it using the
-fully-qualified UDF name (e.g. `<PROJECT>.<DATASET>.<UDF_NAME>()`).
-
-If you are pasting and executing the UDF DDL statement in the BigQuery Console,
-you must modify the UDF name to include your desired dataset, and optionally,
-the project ID:
-
-`CREATE OR REPLACE FUNCTION your_dataset.function_name() . . .`
-
-or
-
-`CREATE OR REPLACE FUNCTION project_id.your_dataset.function_name() . . .`
+## Deploying the UDFs
+
+### Deploy with Cloud Build (Recommended)
+
+1. Authenticate using the Cloud SDK and set the BigQuery project in which you'll
+   deploy your UDF(s):
+
+   ```bash 
+   gcloud init
+   ```
+
+1. Enable the Cloud Build API and grant the default Cloud Build service account
+   the BigQuery Job User and Data Editor roles
+   ```bash
+   gcloud services enable cloudbuild.googleapis.com && \
+   gcloud projects add-iam-policy-binding \
+     $(gcloud config get-value project) \
+     --member=serviceAccount:$(gcloud projects describe $(gcloud config get-value project) --format="value(projectNumber)")"@cloudbuild.gserviceaccount.com" \
+     --role=roles/bigquery.user && \
+   gcloud projects add-iam-policy-binding \
+     $(gcloud config get-value project) \
+     --member=serviceAccount:$(gcloud projects describe $(gcloud config get-value project) --format="value(projectNumber)")"@cloudbuild.gserviceaccount.com" \
+     --role=roles/bigquery.dataEditor
+   ```
+1. Deploy the UDFs by submitting the following:
+
+   ```bash
+   # Deploy to US
+   gcloud builds submit . --config=deploy.yaml --substitutions _BQ_LOCATION=US
+   ```
+   > Note: Deploy to a different location by setting `_BQ_LOCATION` to your own
+   > desired value.
+
+### Deploy with your own machine
+
+Run the following in your machine's terminal to deploy all UDFs in your own
+BigQuery project.
+
+1. Authenticate using the Cloud SDK and set the BigQuery project in which you'll
+   deploy your UDF(s):
+
+   ```bash 
+   gcloud init
+   ```
+
+1. Install the dataform CLI tool:
+
+   ```bash
+   npm i -g @dataform/cli
+   ```
+
+1. Set env variable BQ_LOCATION to the BigQuery location in which you want to
+   deploy the UDFs and then run the `deploy.sh` helper script to deploy the
+   UDFs:
+
+   ```bash
+   # Deploy to US
+   export BQ_LOCATION=US && bash deploy.sh
```

Comment:
I consider the sections independent of each other because users will choose one method of deployment and I wanted all dependencies covered in each section.